luigi/format.py

```diff
@@ -211,37 +237,120 @@ def seekable(self):
         return False
 
 
+class BaseWrapper(object):
+
+    def __init__(self, stream, *args, **kwargs):
+        self._stream = stream
+        try:
+            super(BaseWrapper, self).__init__(stream, *args, **kwargs)
+        except TypeError:
+            pass
+
+    def __getattr__(self, name):
+        if name == '_stream':
+            raise AttributeError(name)
+        return getattr(self._stream, name)
+
+    def __enter__(self):
+        self._stream.__enter__()
+        return self
+
+    def __exit__(self, *args):
+        self._stream.__exit__(*args)
+
+    def __iter__(self):
+        for line in self._stream:
+            yield line
+        self.close()
+
+
+class NewlineWrapper(BaseWrapper):
+
+    def __init__(self, stream, newline=None):
+        if newline is None:
+            self.newline = newline
+        else:
+            self.newline = newline.encode('ascii')
+
+        if self.newline not in (b'', b'\r\n', b'\n', b'\r', None):
+            raise ValueError("newline need to be one of {b'', b'\r\n', b'\n', b'\r', None}")
+        super(NewlineWrapper, self).__init__(stream)
+
+    def read(self, n=-1):
+        b = self._stream.read(n)
+
+        if self.newline == b'':
+            return b
+
+        if self.newline is None:
+            newline = b'\n'
+
+        return re.sub(b'(\n|\r\n|\r)', newline, b)
+
+    def writelines(self, lines):
+        if self.newline is None or self.newline == '':
+            newline = os.linesep.encode('ascii')
+        else:
+            newline = self.newline
+
+        self._stream.writelines(
+            [re.sub(b'(\n|\r\n|\r)', newline, line) for line in lines]
```

Comment:
I tested, it can. It rely on the fact that the underlying stream will be able to handle. I'll change it.